12-say-grpc/backend/vendor/github.com/campoy/justforfunc/13-cgo-flite/flite/flite.go

```diff
@@ -0,0 +1,55 @@
+package flite
+
+// #cgo CFLAGS: -I /usr/include/flite
+// #include <flite.h>
+// #include <stdlib.h>
+// #cgo LDFLAGS: -L /usr/lib/x86_64-linux-gnu -lflite -lflite_cmu_us_kal -lflite_usenglish
+// extern cst_voice *register_cmu_us_kal(const char *voxdir);
+import "C"
+import (
+	"fmt"
+	"io/ioutil"
+	"unsafe"
+)
+
+var voice *C.cst_voice
+
+func init() {
+	C.flite_init()
+	voice = C.register_cmu_us_kal(nil)
+}
+
+func TextToSpeechFile(path, text string) error {
+	if voice == nil {
+		return fmt.Errorf("could not find default voice")
+	}
+
+	ctext := C.CString(text)
+	cpath := C.CString(path)
+	C.flite_text_to_speech(ctext, voice, cpath)
+	C.free(unsafe.Pointer(ctext))
+	C.free(unsafe.Pointer(cpath))
+	return nil
+}
+
+// TextToSpeech synthesizes the text in a string and returns a slice of bytes
+// containing the generated output.
+func TextToSpeechBytes(text string) ([]byte, error) {
+	f, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, fmt.Errorf("could not create tmp file: %v", err)
+	}
+	if err := f.Close(); err != nil {
```

Comment:
Not necessarily, but it needs to be closed at some point, right?
I'll change it to f.Close() since I don't really think this might ever fail